engine/executor/server.go

```diff
@@ -220,19 +229,46 @@ func (s *Server) makeTask(
 	return taskutil.WrapWorker(newWorker), nil
 }
 
+// precheckMasterMeta checks job master metadata before running it, stop task
+// creating if job master has met a business unretryable error.
+// Return error means meets failure in this function or job creation should be
+// terminated.
+func precheckMasterMeta(
```

Comment:
Can we split the `IsRetryableError` from `registry` to make it more common?
Since we have the `GracefulExit` mechanism, we can check the busniess error in `jobmanager` to decide whether we need to redispatch this jobmaster.